src/react-a11y-dialog.tsx

```diff
@@ -16,22 +15,79 @@ const useIsMounted = () => {
 }
 
 export type ReactA11yDialogProps = {
+  /**
+   * The `role` attribute of the dialog element, either `dialog` (default) or
+   * `alertdialog` to make it a modal (preventing closing on click outside of
+   * ESC key).
+   */
   role?: 'dialog' | 'alertdialog'
+  /**
+   * The HTML `id` attribute of the dialog element, internally used by
+   * a11y-dialog to manipulate the dialog.
+   */
   id: string
+  /**
+   * The title of the dialog, mandatory in the document to provide context to
+   * assistive technology. Could be hidden (while remaining accessible) with
+   * CSS though.
+   */
   title: React.ReactNode
+  /**
+   * A ref callback called when the component has mounted, receiving the instance
+   * of A11yDialog so that it can be programmatically accessed later on.
+   * 
+   * @example
+   * const [dialogElement, setDialogElement] = React.useState(null);
+   * // ...
+   * dialogRef={setDialogElement}
```

Comment:
`useRef` returns an object so it doesn't work, so using `useState` instead

Reply:
Not necessarily. The way it is typically used is like so:

```jsx
const dialogRef = useRef()

<A11yDialog dialogRef={instance => (dialogRef.current = instance)} />
```

Reply:
hmm yeah, but this won't cause a rerender. updated though :)

Reply:
Which is fine, it‘s just to get the bindings. Never had a problem with this. :)